ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -336,6 +338,20 @@ func (r *RayClusterReconciler) reconcilePods(instance *rayiov1alpha1.RayCluster)
 	if err := r.List(context.TODO(), &headPods, client.InNamespace(instance.Namespace), filterLabels); err != nil {
 		return err
 	}
+	if EnableBatchScheduler {
+		var minMember int32
```

Comment:
It would be great to pull as much of the added line count out into another file.
One idea would be to shift the resource logic into `r.syncPodGroup`. The implementation of syncPodGroup could be kept somewhere else.

That would help communicate the third-party/optional nature of this feature and would also keep the code a bit tidier.

Reply:
Hey @DmitriGekhtman, thanks for the feedback. I adapted the abstraction from the Spark Operator and moved all code into a separate batchscheduler directory, which hides the implementation behind an interface. Let me know what you think.

Reply:
LGTM!